src/structures/interaction/CommandInteraction.ts

```diff
@@ -0,0 +1,60 @@
+import Collection from '@discordjs/collection';
+import {
+	APIApplicationCommandGuildInteraction,
+	APIApplicationCommandInteractionDataOption,
+	Snowflake,
+} from 'discord-api-types';
+import { Client } from '../../clients';
+import { GuildChannel } from '../channel/GuildChannel';
+import { GuildMember } from '../guild/GuildMember';
+import { Role } from '../Role';
+import { User } from '../User';
+import { Interaction } from './Interaction';
+
+export class CommandInteraction extends Interaction {
+	public id!: Snowflake;
+	public name!: string;
+	public options?: APIApplicationCommandInteractionDataOption[];
+	public users = new Collection<Snowflake, User>();
+	public roles = new Collection<Snowflake, Role>();
+	public members = new Collection<Snowflake, GuildMember>();
+	public channels = new Collection<Snowflake, GuildChannel>();
+
+	public constructor(client: Client, data: APIApplicationCommandGuildInteraction) {
+		super(client, data);
+
+		this.$patch(data);
+	}
+
+	public $patch(data: APIApplicationCommandGuildInteraction): void {
+		super.$patch(data);
+
+		if (data.data.resolved?.users) {
+			for (const [id, user] of Object.entries(data.data.resolved.users)) {
+				this.users.set(id as Snowflake, new User(this.client, user));
+			}
+		}
+
+		/*if (data.data.resolved?.roles) {
```

Comment:
🤔

Reply:
We need an instance of Guild for that using the property guidId of Interaction